pkg/destroy/aws/shared.go

```diff
@@ -181,7 +181,6 @@ func (o *ClusterUninstaller) cleanSharedRoute53(ctx context.Context, session *se
 	if err != nil {
 		return err
 	}
-	logger = logger.WithField("id", id)
```

Comment:
I think that we should keep this so that the log events such as "Cleaned record sets from hosted zone" include the ID of the hosted zone. It is not a huge deal, because the log events will already have the ARN of the hosted zone, but I think it is a bit nicer to be consistent by having the "id" field.

In the `if publicZoneID != ""` section of `cleanSharedHostedZone`, create a new logger with the "id" field set to the public hosted zone for use in deleting the matching public record.
```
publicZoneLogger := logger.WithField("id", id)
```